src/java-opentimelineio/src/main/java/io/opentimeline/opentimelineio/Any.java

```diff
@@ -0,0 +1,242 @@
+            this.initTimeRange((TimeRange) value);
+            anyTypeClass = TimeRange.class.getCanonicalName();
+        } else if (value instanceof TimeTransform) {
+            this.initTimeTransform((TimeTransform) value);
+            anyTypeClass = TimeTransform.class.getCanonicalName();
+        } else if (value instanceof AnyDictionary) {
+            this.initAnyDictionary((AnyDictionary) value);
+            anyTypeClass = AnyDictionary.class.getCanonicalName();
+        } else if (value instanceof AnyVector) {
+            this.initAnyVector((AnyVector) value);
+            anyTypeClass = AnyVector.class.getCanonicalName();
+        } else if (value instanceof SerializableObject) {
+            this.initSerializableObject((SerializableObject) value);
+            anyTypeClass = SerializableObject.class.getCanonicalName();
+        } else throw new RuntimeException("Any: Type not supported.");
+    }
+
+    private void initBool(boolean b) {
+        this.initializeBool(b);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initInt(int i) {
+        this.initializeInt(i);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initLong(long a) {
+        this.initializeLong(a);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initDouble(double d) {
+        this.initializeDouble(d);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initString(String string) {
+        this.initializeString(string);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initRationalTime(RationalTime rationalTime) {
+        this.initializeRationalTime(rationalTime);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initTimeRange(TimeRange timeRange) {
+        this.initializeTimeRange(timeRange);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initTimeTransform(TimeTransform timeTransform) {
+        this.initializeTimeTransform(timeTransform);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initAnyVector(AnyVector anyVector) {
+        this.initializeAnyVector(anyVector);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initAnyDictionary(AnyDictionary anyDictionary) {
+        this.initializeAnyDictionary(anyDictionary);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private void initSerializableObject(SerializableObject serializableObject) {
+        this.initializeSerializableObject(serializableObject);
+        this.nativeManager.className = this.getClass().getCanonicalName();
+    }
+
+    private native void initializeBool(boolean a);
```

Comment:
For primitive types what does native handle provide?

Reply:
I'm sorry I don't understand the question.

Reply:
@margant could you please clarify your question?

Reply:
sure, wanted to make sure I am not missing anything here, but it looks like we're initializing a boolean which is a primitive type. Is there a need to proxy boolean from C++? I thought we're doing it for higher level types.

Reply:
These native functions aren't a proxy for higher level types. The non native function sets the `className` variable in addtion to initializing the native object.